hbase-server/src/main/java/org/apache/hadoop/hbase/master/balancer/StochasticLoadBalancer.java

```diff
@@ -1236,11 +1236,13 @@ protected double cost() {
         DEFAULT_PRIMARY_REGION_COUNT_SKEW_COST));
     }
 
+    @Override
+    boolean isNeeded() {
+      return cluster.hasRegionReplicas;
+    }
+
     @Override
     protected double cost() {
-      if (!cluster.hasRegionReplicas) {
```

Comment:
We should leave these lines here. `isNeeded` is only called by the `needsBalance` call site. But `cost` will still be called when a balance is determined to be needed.

Reply:
it should follow same logic as `RegionReplicaHostCostFunction`

Reply:
Done